chain-adapter-integration-test/src/integration-test/kotlin/integration/chainadapter/ChainAdapterIntegrationTest.kt

```diff
@@ -0,0 +1,99 @@
+package integration.chainadapter
+
+import com.github.kittinunf.result.failure
+import integration.chainadapter.environment.ChainAdapterIntegrationTestEnvironment
+import mu.KLogging
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import sidechain.iroha.ReliableIrohaChainListener
+import util.getRandomString
+import java.util.*
+
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class ChainAdapterIntegrationTest {
+
+    private lateinit var environment: ChainAdapterIntegrationTestEnvironment
+
+    @BeforeEach
+    fun setUp() {
+        environment = ChainAdapterIntegrationTestEnvironment()
+        environment.adapter.init().failure { ex -> throw ex }
+    }
+
+    @AfterEach
+    fun tearDown() {
+        environment.close()
+    }
+
+    /**
+     * Note: Iroha must be deployed to pass the test.
+     * @given running chain-adapter
+     * @when new transactions appear in Iroha blockchain
+     * @then RabbitMQ consumer reads new transactions
+     * in the same order as they were published
+     */
+    @Test
+    fun testRuntimeBlocksWerePublished() {
+        val transactions = 10
+        val queueName = String.getRandomString(5)
```

Comment:
Actually, if you do not specify queue name, rmq will do that for you and return it as queueDeclare result

Reply:
But queue name is part of `ReliableIrohaChainListener` constructor.

Reply:
We can think about making it optional, but maybe later